boot/systems.go

```diff
@@ -342,3 +343,99 @@ func InspectTryRecoverySystemOutcome(dev Device) (outcome TryRecoverySystemOutco
 
 	return outcome, trySystem, nil
 }
+
+// MarkRecoverySystemGood marks the provided recovery system as a good one, and
+// esures that the system is present in the list of good recovery systems and
+// current recovery systems in modeenv. If the system uses encryption, the keys
+// will updated state. If resealing fails, an attempt to restore the previous
+// state is made
+func MarkRecoverySystemGood(dev Device, systemLabel string) (err error) {
```

Comment:
we discussed a different name and signature for this in the context of the next PR

Reply:
The API has been tweaked as discussed.